jerry-core/ecma/operations/ecma-regexp-object.h

```diff
@@ -29,26 +29,30 @@
  * @{
  */
 
-#define RE_EXECUTE_RECURSION_LIMIT  1000  /* Limit of RegExp executor recursion depth */
-#define RE_EXECUTE_MATCH_LIMIT      10000 /* Limit of RegExp execetur matching steps */
+/**
+ * Limit of RegExp executor recursion depth
+ */
+#define RE_EXECUTE_RECURSION_LIMIT  1000
+
+/**
+ * Limit of RegExp execetur matching steps
+ */
+#define RE_EXECUTE_MATCH_LIMIT      10000
 
 /**
  * RegExp executor context
- *
- * FIXME:
- *       Add comments with description of the structure members
  */
 typedef struct
 {
-  const lit_utf8_byte_t **saved_p;
-  const lit_utf8_byte_t *input_start_p;
-  const lit_utf8_byte_t *input_end_p;
-  uint32_t match_limit;
-  uint32_t recursion_depth;
-  uint32_t num_of_captures;
-  uint32_t num_of_non_captures;
-  uint32_t *num_of_iterations;
-  uint8_t flags;
+  const lit_utf8_byte_t **saved_p; /**< saved result string pointers */
```

Comment:
It is also described in `ecma-regexp-object.cpp`.

```
/**
 * RegExp results are stored in an array of string pointers. If N is the number
 * of groups then the length of the array is 2*N, because every group has a start
 * and end. We have to handle those pointers.
 *
 *   [0] RE global start
 *   [1] RE global end
 *   [2] 1st group start
 *   [3] 1st group end
 *   ...
 *   [n]   n/2 th group start
 *   [n+1] n/2 th group end
 */
```
It is basically the State of RegExp. I added reference to `ECMA 262 v5, 15.10.2.1, State`.